src/lib/kadm5/kadm_rpc.h

```diff
@@ -108,6 +108,15 @@ struct setkey3_arg {
 };
 typedef struct setkey3_arg setkey3_arg;
 
+struct setkey4_arg {
+	krb5_ui_4 api_version;
+	krb5_principal princ;
+	krb5_boolean keepold;
+	kadm5_key_data *key_data;
+	int n_key_data;
+};
```

Comment:
I strongly encourage you to add that one argument (do/don't change the principal's current kvno).

Also, will it be the case that the `key_data` argument can contain keys of unspecified kvno to mean "add this as the next kvno"?